install_config/storage_examples/storage_classes_dynamic_provisioning.adoc

```diff
@@ -0,0 +1,322 @@
+   requests:
+     storage: 5Gi
+----
+====
+
+====
+----
+
+# oc create -f default-pvc.yaml 
+persistentvolumeclaim "pvc-engineering2" created
+                                                                   3s
+[root@screeley-sc1 gce]# oc get pvc
+NAME               STATUS    VOLUME                                     CAPACITY   ACCESSMODES   AGE
+pvc-engineering    Bound     pvc-e9b4fef7-8bf7-11e6-9962-42010af00004   10Gi       RWX           41m
+pvc-engineering2   Bound     pvc-a9f70544-8bfd-11e6-9962-42010af00004   5Gi        RWX           7s
+
+----
+====
+
+
+As a cluster-admin or storage-admin view the Persistent Volumes that are defined so far
+
+====
+----
+
+# oc get pv
+NAME                                       CAPACITY   ACCESSMODES   RECLAIMPOLICY   STATUS    CLAIM                     REASON    AGE
+pvc-a9f70544-8bfd-11e6-9962-42010af00004   5Gi        RWX           Delete          Bound     rh-eng/pvc-engineering2             5m <1>
+pvc-ba4612ce-8b4d-11e6-9962-42010af00004   5Gi        RWO           Delete          Bound     mytest/gce-dyn-claim1               21h
+pvc-e9b4fef7-8bf7-11e6-9962-42010af00004   10Gi       RWX           Delete          Bound     rh-eng/pvc-engineering              46m <2>
+
+----
+<1> This PV was bound to our _default_ dynamic volume from the _default_ StorageClass.
+<2> This PV was bound to our first PVC from example 1 with our _fast_ StorageClass.
+====
+
+Create a manual disk in the GCE console (not dynamically provisioned) and a Persistent Volume that connects to the new GCE disk.
+
+Now view the PVs again, notice we have a pv-manual-gce volume Available.
+
+====
+----
+
+# oc get pv
+NAME                                       CAPACITY   ACCESSMODES   RECLAIMPOLICY   STATUS      CLAIM                     REASON    AGE
+pv-manual-gce                              35Gi       RWX           Retain          Available                                       4s
+pvc-a9f70544-8bfd-11e6-9962-42010af00004   5Gi        RWX           Delete          Bound       rh-eng/pvc-engineering2             12m
+pvc-ba4612ce-8b4d-11e6-9962-42010af00004   5Gi        RWO           Delete          Bound       mytest/gce-dyn-claim1               21h
+pvc-e9b4fef7-8bf7-11e6-9962-42010af00004   10Gi       RWX           Delete          Bound       rh-eng/pvc-engineering              53m
+
+----
+====
+
+Now create another claim identical to the PVC above but change the name (no annotation set). Will the claim get satisfied by the _default_ StorageClass or the manually available Volume?
+
+
+.Claim Spec Definition
+====
+[source,yaml]
+----
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+ name: pvc-engineering3
+spec:
+ accessModes:
+  - ReadWriteMany
+ resources:
+   requests:
+     storage: 15Gi
+
+----
+====
+
+
+Even though the claim was satisfied by the existing and available _pv-manual-gce_ PV, because _default_ StorageClass was enabled, a volume was dynamically provisioned based on the claim request.
```

Comment:
Had the manual PV been created specifying the storageclass was `default` it owuld have been claimed and nothing would have been provisioned.

So the claim was not in fact satisfied by the manual PV.